src/cascadia/TerminalCore/Terminal.hpp

```diff
@@ -156,12 +156,17 @@ class Microsoft::Terminal::Core::Terminal final :
 
 #pragma region TextSelection
     // These methods are defined in TerminalSelection.cpp
+    enum class SelectionExpansionMode
+    {
+        Cell,
+        Word,
+        Line
+    };
     const bool IsCopyOnSelectActive() const noexcept;
-    void DoubleClickSelection(const COORD position);
-    void TripleClickSelection(const COORD position);
+    void MultiClickSelection(const COORD viewportPos, SelectionExpansionMode expansionMode);
     void SetSelectionAnchor(const COORD position);
-    void SetEndSelectionPosition(const COORD position);
-    void SetBoxSelection(const bool isEnabled) noexcept;
+    void SetSelectionEnd(const COORD position, std::optional<SelectionExpansionMode> expansionMode = std::nullopt);

```

Comment:
is nullopt treated equivalently to Cell? If so, set cell as the default and ditch the optional

Reply:
Nope.
- `nullopt` --> use the current _multiClickSelectionMode
- `<new mode>` --> overwrite it, and use that

The overall plan is that this'll be even more useful for #4557. We can call `SetSelectionEnd(terminalPosition, ::Terminal::SelectionExpansionMode::Line);` to only expand one of the selection anchors to a line, as opposed to both anchors.